components/ProfilePage/AvatarSelector.js

```diff
@@ -0,0 +1,254 @@
+            test: 'test',
+          },
+        }}
+        onChange={e => onChange(e.target.value)}
+        disabled={options.length === 0}
+      >
+        {options.map(value => (
+          <MenuItem key={value} value={value} className={classes.menuItem}>
+            <svg className={classes.avatarPiece} viewBox={viewBox}>
+              {value && pieceType && createElement(pieceType[value])}
+            </svg>
+          </MenuItem>
+        ))}
+      </TextField>
+    </>
+  );
+}
+
+function ColorPicker({ options, anchorEl, selected, onChange = () => {} }) {
+  const [expand, setExpand] = useState(false);
+  const classes = useStyles({ expand });
+  return (
+    <>
+      <TextField
+        select
+        classes={{ root: classes.colorPickerRoot }}
+        InputProps={{
+          classes: {
+            root: classes.inputClass,
+          },
+          disableUnderline: true,
+        }}
+        SelectProps={{
+          classes: {
+            root: classes.selectClass,
+          },
+          onOpen: () => setExpand(true),
+          onClose: () => setExpand(false),
+          value: selected,
+          MenuProps: {
+            PopoverClasses: {
+              paper: classes.popoverRoot,
+            },
+            anchorOrigin: {
+              vertical: 'bottom',
+              horizontal: 'center',
+            },
+            transformOrigin: {
+              vertical: 'center',
+              horizontal: 'center',
+            },
+            anchorEl,
+          },
+          SelectDisplayProps: {
+            test: 'test',
+          },
+        }}
+        onChange={e => onChange(e.target.value)}
+        disabled={options.length === 0}
+      >
+        {options.map(value => (
+          <MenuItem key={value} value={value} className={classes.menuItem}>
+            <span
+              className={classes.colorGird}
+              style={{ background: value }}
+            ></span>
+          </MenuItem>
+        ))}
+      </TextField>
+    </>
+  );
+}
+
+const pieces = [
+  {
+    pieceType: Accessories,
+    type: 'accessory',
+    viewBox: '-75 -125 500 400',
+    title: t`Accessory`,
+  },
+  {
+    pieceType: BustPose,
+    type: 'body',
+    viewBox: '0 150 1200 1200',
+    title: t`Body`,
+  },
+  {
+    pieceType: Face,
+    type: 'face',
+    viewBox: '0 -20 300 400',
+    title: t`Face`,
+  },
+  {
+    pieceType: Hair,
+    type: 'hair',
+    viewBox: '0 -100 550 750',
+    title: t`Hair`,
+  },
+  {
+    pieceType: FacialHair,
+    type: 'facialHair',
+    viewBox: '-50 -100 500 400',
+    title: t`Facial Hair`,
+  },
+];
+
+function AvatarSelector({ avatarData, onChange }) {
+  const wrapperEl = useRef(null);
+  const classes = useStyles();
+
+  return (
+    <div ref={wrapperEl} className={classes.wrapper}>
+      {pieces.map(({ pieceType, type, viewBox, title }) => (
+        <ComponentInput
+          key={type}
+          pieceType={pieceType}
+          anchorEl={wrapperEl?.current}
```

Comment:
When `ComponentInput` is rendered, `wrapperEl.current` is null; after `AvatarSelector` is rendered, `wrapperEl` is populated, but that does not trigger re-render.

`anchorEl` [supports a function](https://material-ui.com/api/menu/). I think when a function is passed, material-ui will invoke it when menu is opened, thus this can fetch the latest ref value in scope.

```suggestion
          anchorEl={() => wrapperEl.current}
```

Ref object returned by `useRef` is always an object that has `currrent` property (even just after initialization, it's `{current: null}`) so no need for `?.`.

The same applies to `<ColorPicker>` below.